source/administration/config-in-database.rst

```diff
@@ -10,24 +10,175 @@ Benefits to using this option:
+
+Here are two example connection strings:
+
+*MySQL*
+
+.. code-block:: text
+
+   mysql://mmuser:really_secure_password@tcp(127.0.0.1:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s
+
+*PostgreSQL*
+
+.. code-block:: text
+
+   postgres://mmuser:really_secure_password@localhost:5432/mattermost?sslmode=disable&connect_timeout=10
+
+2. Create Environment File:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**NOTE:** If you are running Mattermost in an HA cluster this step must be done on **all** servers in the cluster. 
+
+Create the file  ``/opt/mattermost/config/mattermost.environment``  to set the ``MM_CONFIG`` environment variable to the database connection string. For example:
+
+*MySQL*
+
+.. code-block:: text
+
+   MM_CONFIG='mysql://mmuser:mostest@tcp(127.0.0.1:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s'
+
+*PostgreSQL*
+
+.. code-block:: text
+
+   MM_CONFIG='postgres://mmuser:mostest@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10'
+
+**NOTE:** Be sure to escape an single quotes in the database connection string by placing a ``\`` in front of them like this ``\'``. For example:
+
+.. code-block:: text
+
+   MM_CONFIG='mysql://mmuser:it\'s-a-password!@tcp(127.0.0.1:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s'
+
+Finally, run this command to verify the permissions on your Mattermost directory:
+
+.. code-block:: bash
+
+   sudo chown -R mattermost:mattermost /opt/mattermost
+
+3. Modify Mattermost systemd file
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, find the ``mattermost.service`` file by first running this command:
+
+.. code-block:: bash
+
+   sudo systemctl status mattermost.service
+
+The second line of output will have the location of the running ``mattermost.service``
+
+.. code-block:: text
+
+      Loaded: loaded (/lib/systemd/system/mattermost.service; enabled; vendor preset: enabled)
+
+Edit this file as ``root`` to add this line just before the line that beings with ``ExecStart``\ :
+
+.. code-block:: text
+
+   EnvironmentFile=/opt/mattermost/config/mm.environment
+
+Here is a complete ``mattermost.service`` file with the ``EnvironmentFile`` line added:
+
+.. code-block:: text
+
+   [Unit]
+   Description=Mattermost
+   After=network.target
+   After=mysql.service
+   Requires=mysql.service
+
+   [Service]
+   Type=notify
+   EnvironmentFile=/opt/mattermost/config/mm.environment
+   ExecStart=/opt/mattermost/bin/mattermost
+   TimeoutStartSec=3600
+   Restart=always
+   RestartSec=10
+   WorkingDirectory=/opt/mattermost
+   User=mattermost
+   Group=mattermost
+   LimitNOFILE=49152
+
+   [Install]
+   WantedBy=mysql.service
+
+**NOTE:** If you are using PostgreSQL as your database the ``mysql.service`` **must** be replaced with ``postgresql.service``. The easiest way to avoid making a mistake is to just add the ``EnvironmentFile`` line and not copy the entire example.
```

Comment:
```suggestion
.. note::
  If you are using PostgreSQL as your database the ``mysql.service`` must be replaced with ``postgresql.service``. The easiest way to avoid making a mistake is to just add the ``EnvironmentFile`` line and not copy the entire example.
```